### PR TITLE
listpeers enhancements and suppress duplicates

### PR DESCRIPTION
**Pull request by rustyrussell**
I had these lying around; I'm cleaning up my branches so I figured it was time to push them.

### CODE REVIEW

**Comment by cdecker** [comment]
Looks quite ok, except maybe the last commit. I don't get the logic behind deferring a list of updates when we can combine multiple updates into a single one. If we change the `base_fee` in update 1 and the `proportional_fee` in update 2, we can just create a new update that combines both changes, no need to actually queue two of them.

**Comment by rustyrussell** [comment]
@cdecker I don't understand.  Updates already override, and don't combine.  So update 3 will always replace update 2.  (Not that we have a mechanism to change the fees we advertise yet). 

The logic in the last patch is simple: first, suppress obviously redundant updates.  Second, wait a while in case we see a redundant update, eg. disabling then re-enabling channel.

**Comment by cdecker** [comment]
I don't really see the point of making it a list. Shouldn't we only ever have 1 pending update. If we already have an update waiting, and want to queue a new one, then we should amend the existing one, instead of queuing a new one.

**Comment by rustyrussell** [comment]
We delete the previous pending one, and append a new one.  It does mean we restart the 15 second timer, though?

**Comment by rustyrussell** [comment]
OK, this is simpler, and nicer IMHO.

**Comment by jb55** [comment]
bisect shows compile error on f9b8237d500c0a2f8e8de171ba287760facda724 due to structeq update conflict?

```
gossipd/gossip.c: In function ‘find_local_update’:
gossipd/gossip.c:1702:7: error: implicit declaration of function ‘structeq’; did you mean ‘strtoq’? [-Werror=implicit-function-declaration]
   if (structeq(scid, &i->scid))
       ^~~~~~~~
       strtoq
cc1: all warnings being treated as errors
```

**Comment by cdecker** [comment]
Do we really need to re-enable the rebase check on github?

File: lightningd/peer_control.c [hunk 1]
```diff
@@ -776,10 +777,18 @@ static void gossipd_getpeers_complete(struct subd *gossip, const u8 *msg,
 			json_array_end(response);
 		}
 
-		json_add_node_decoration(response, nodes, &p->id);
+		/* Search gossip reply for this ID, to add extra info. */
+		for (size_t i = 0; i < tal_len(nodes); i++) {
+			if (pubkey_eq(&nodes[i]->nodeid, &p->id)) {
```
Comment (jb55) [inline lightningd/peer_control.c]: I'm getting an invalid pointer here, so I crash in listpeers now. will poke around to try and figure out why...

**Pull request merged**